components/region_cache_memory_engine/src/write_batch.rs

```diff
@@ -396,11 +396,17 @@ impl RangeCacheWriteBatchEntry {
     }
 
     pub fn calc_put_entry_size(key: &[u8], value: &[u8]) -> usize {
-        key.len() + value.len() + ENC_KEY_SEQ_LENGTH + 2 * MEM_CONTROLLER_OVERHEAD /* one for key and one for value */
+        RangeCacheWriteBatchEntry::memory_size_required_for_key_value(key, value)
     }
 
     pub fn cal_delete_entry_size(key: &[u8]) -> usize {
-        key.len() + ENC_KEY_SEQ_LENGTH
+        // delete also has value which is an empty bytes
+        RangeCacheWriteBatchEntry::memory_size_required_for_key_value(key, b"")
```

Comment:
Recommend introducing a const for the value of delete entries.

Reply:
Done